### PR TITLE
feat : Watch later

### PR DESCRIPTION
**Pull request by arpit01923**
Description-

- watchLater page functionality should be added
- contains a api data
- add components navbar, main, footer
- click on watchLater option will add video to watchLater page

File to check-

- src/contexts/watch-later.jsx
- src/reducer/watch-later.jsx
- src/utils/watch-later.jsx

Preview-

https://user-images.githubusercontent.com/86866251/164971832-15f990fc-053b-49c6-8cb3-5522b66e2749.mp4

### CODE REVIEW

**Review by ranitasahaa1230 — APPROVED** [review]
Nice and clean code,
Good naming conventions followed

File: src/contexts/watch-later.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const WatchLaterContext = createContext();
+
+const WatchLaterProvider = ({ children }) => {
+
+  const [watchLaterState, watchLaterDispatch] = useReducer(watchLaterReducer, {
+    watchLater: [],
+  });
+
+  console.log(watchLaterState.watchLater);
```
Comment (Aditi0505) [inline src/contexts/watch-later.jsx]: Remove console.log

File: src/pages/video-listing/video-card.jsx [hunk 2]
```diff
@@ -1,37 +1,68 @@
...
+export const VideoCard = ({ filtervideo }) => {
+  const [show, setShow] = useState(false);
+  const {watchLaterDispatch} = useWatchLater();
+
+  const cardPopUp = () => {
+    setShow((prev) => !prev);
+  };
+
+  const addWatchLater = (video) => {
+    watchLaterDispatch({ type: "Add to watch-later", payload: video });
```
Comment (Aditi0505) [inline src/pages/video-listing/video-card.jsx]: Suggestion: The convention is to use Uppercase letters in type.

File: src/reducer/watch-later.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  switch (watchLaterAction.type) {
+    case "Add to watch-later":
+      return {
+        ...watchLaterState,
+        watchLater: addWatchLaterVideos(
+          watchLaterState.watchLater,
+          watchLaterAction.payload
+        ),
+      };
+    case "Delete from watch-later":
```
Comment (Aditi0505) [inline src/reducer/watch-later.jsx]: Would suggest you to change this to DELETE_FROM_WATCH_LATER or DELETE_VIDEO

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions.

1. Clean code 
2. Good folder structure.

**Pull request merged**